### DISCUSSION TITLE
Add a heat source with a specified heat flux

### DISCUSSION THREAD

**Discussion by RSuryaNarayan** [AMReX-Combustion/PeleC#384] [Q&A] [answered]
Hello, 

Is there a way to add a heat flux of specified intensity over a region in the domain? This must ideally add a source term to the energy equation and proceed with the solution. 

Thanks a lot

**Comment by whitmanscu** [comment, marked as answer]
I've done this before via modifying `PeleC::fill_ext_source` in `External.cpp` to add energy. You then need to specify `pelec.add_ext_src = 1` in the inputs file. 

I will caution that I've found ignition via this method to be less stable than heat transfer from an EB surface. It's been a while (over a year) since I've tried it with an external source and it may be more stable at this time. If you are looking to ignite a mixture via this method, I'd be curious to hear how it works for you.

**Comment by RSuryaNarayan** [comment]
Thanks for your reply. 

If I am not wrong, I should be filling in the appropriate ```n``` in the following line with the value of the heat flux in CGS units?:
https://github.com/AMReX-Combustion/PeleC/blob/6749e57f5464907e9cc1cae68871f4494f61be8d/Source/External.cpp#L83

something like ```n==UTEMP`` then Farr(i, j, k, n) = some value```

**Reply by drummerdoc** [reply]
You'd want to use ``n=UEDEN`` I believe

**Reply by RSuryaNarayan** [reply]
Thanks for the clarification!

**Comment by whitmanscu** [comment]
Basically, yes, although I believe you want to add to `UEINT` and `UEDEN` rather than `UTEMP`,  for proper use in reactions and since temperature is then recomputed from internal energy. Whatever you add is effectively the `S_ext` term at the end of the RHS in the equations here: https://pelec.readthedocs.io/en/latest/Equations.html.

Also if it's not clear, you can use `i, j, k` in conjunction with the commented-out `dx` and `prob_lo` to specify where you want to apply the flux. It can also be time-varying, relative to the old and new states, etc. (with the new state typically being correct to use based on where the term is added in `Advance.cpp`.

**Reply by RSuryaNarayan** [reply]
yes it makes sense now. Similar to what we find in ```pc_initdata```. Thanks a lot for the prompt replies. @whitmanscu and @drummerdoc !
